src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = '';
+
+    for (let element = 0; element < this.length; element++) {
+      if ((this[element] !== null) && (this[element] !== undefined)) {
+        joinedStr += this[element];
+      }
+
+      if (element !== (this.length - 1)) {
```

Comment:
you don't need extra brackets in if
```suggestion
      if (this[element] !== null && this[element] !== undefined) {
        joinedStr += this[element];
      }

      if (element !== this.length - 1) {
```